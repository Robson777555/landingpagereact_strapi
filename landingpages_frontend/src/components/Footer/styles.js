import styled, { css } from 'styled-components';
import { Container as TextComponent } from '../../components/TextComponent/styles';
import { Container as SectionContainer } from '../SectionContainer/styles';

export const Container = styled.footer`
  /* eslint-disable-next-line no-unused-vars */
  ${({ theme }) => css`
    font-family:${theme.fonts.family.default};
    text-align: center;
    font-size: ${theme.fonts.sizes.small};
    border-top: 0.1rem solid ${theme.colors.mediumGray};
    
    a{
      color: inherit;
      text-decoration: none;
      
    }
    & ${TextComponent} {
      font-size: ${theme.fonts.sizes.small};
    }

    & ${SectionContainer} {
      padding-top: 0;
      padding-bottom: 0;
    }
  `}
  
`;
