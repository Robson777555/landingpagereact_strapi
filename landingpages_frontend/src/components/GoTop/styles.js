import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    position: fixed;
    background: ${theme.colors.primaryColor};
    color: ${theme.colors.white};
    display: flex;
    align-items: center;
    justify-content: center;
    width: 4rem;
    height: 4rem;
    bottom: 2rem;
    right: 2rem;
    border-radius: 50%;
    z-index: 6;
    opacity: 0.7;
    transition: all 0.3s ease;
    cursor: pointer;

    /* Melhorias para mobile */
    @media (max-width: 768px) {
      width: 3.5rem;
      height: 3.5rem;
      bottom: 1.5rem;
      right: 1.5rem;
    }

    &:hover {
      opacity: 1;
      transform: scale(1.05);
    }

    /* Garante que seja clic√°vel em mobile */
    -webkit-tap-highlight-color: transparent;
    touch-action: manipulation;
  `}
`;
