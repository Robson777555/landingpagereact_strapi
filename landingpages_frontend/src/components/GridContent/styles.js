import styled, { css } from 'styled-components';

export const Container = styled.div`
  /* eslint-disable-next-line no-unused-vars */
  ${({ theme }) => css`
    font-family:${theme.fonts.family.default}; 
    text-align: center;
    max-width: 58rem;
    margin: 0 auto;
    

  `}
`;

export const html = styled.div`
  /* eslint-disable-next-line no-unused-vars */
  ${({ theme }) => css`
    font-family:${theme.fonts.family.default}; 
    margin:${theme.spacings.xhuge} 0;
    text-align: center;
    
    pre {
      display: inline-block;
      
      margin: 0 auto;
    }

  `}
`;
