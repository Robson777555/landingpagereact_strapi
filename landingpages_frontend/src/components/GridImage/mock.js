export default {
  background: false,
  title: 'My grid 2',
  description: 'Tecnologias usadas na Landing Page:',
  grid: [
    {
      altText: 'Teste 1',
      srcImg:
        'https://storage.googleapis.com/medium-feed.appspot.com/images%2F9353691196%2F2f49650ed42b1-Para-que-e-usado-o-JavaScript.jpg?r=1',
    },
    {
      altText: 'Teste 2',
      srcImg:
        'https://devsagaz.com.br/content/images/2023/05/Ekran-Resmi-2019-11-18-18.08.13.png?r=2',
    },
    {
      altText: 'Teste 3',
      srcImg:
        'https://mms.businesswire.com/media/20200520005179/en/792790/23/Logo.WhiteBackground.jpg?r=3',
    },
    {
      altText: 'Teste 4',
      srcImg:
        'https://upload.wikimedia.org/wikipedia/commons/d/d9/Node.js_logo.svg?r=4',
    },
    {
      altText: 'Teste 5',
      srcImg:
        'https://blog.codapp.com.br/wp-content/uploads/2022/05/capa.jpg?r=5',
    },
    {
      altText: 'Teste 6',
      srcImg:
        'https://learn.microsoft.com/pt-br/shows/visual-studio-code/media/visual-studio-code-banner-image.jpg?r=6',
    },
  ],
};
