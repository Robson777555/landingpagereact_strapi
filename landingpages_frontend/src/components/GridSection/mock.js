export default {
  title: 'My grid',
  description: 'Esse site é para estudos de react e técnicas de programação',
  background: false,
  grid: [
    {
      title: 'Teste 1',
      description:
        'Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Vestibulum tortor quam, feugiat vitae, ultricies eget, tempor sit amet, ante. Donec eu libero sit amet quam egestas semper. Aenean ultricies mi vitae est. Mauris placerat eleifend leo.',
    },
    {
      title: 'Teste 2',
      description:
        'Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Vestibulum tortor quam, feugiat vitae, ultricies eget, tempor sit amet, ante. Donec eu libero sit amet quam egestas semper. Aenean ultricies mi vitae est. Mauris placerat eleifend leo.',
    },
    {
      title: 'Teste 3',
      description:
        'Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Vestibulum tortor quam, feugiat vitae, ultricies eget, tempor sit amet, ante. Donec eu libero sit amet quam egestas semper. Aenean ultricies mi vitae est. Mauris placerat eleifend leo.',
    },
  ],
};
