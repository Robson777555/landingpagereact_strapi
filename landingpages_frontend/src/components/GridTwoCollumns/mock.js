export default {
  title: 'Grid Two Collumns',
  text: 'As três tecnolgias básicas para o front end são essas.',
  srcImg: 'public/assets/images/javascript.svg',
  background: false,
};
