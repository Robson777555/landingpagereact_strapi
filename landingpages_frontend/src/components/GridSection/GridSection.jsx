import P from 'prop-types';
import * as Styled from './styles';
import { SectionBackground } from '../SectionBackground/SectionBackground';
import { Heading } from '../Heading/Heading';
import { TextComponent } from '../TextComponent/TextComponent';

export const GridSection = ({
  title,
  description,
  grid,
  background = false,
}) => {
  return (
    <SectionBackground colorDark={background}>
      <Styled.Container>
        <Heading size="huge" uppercase colorDark={!background} as="h2">
          {title}
        </Heading>
        <TextComponent>{description}</TextComponent>
        <Styled.Grid>
          {grid.map((el) => (
            <Styled.GridElement key={el.title}>
              <Heading size="medium" colorDark={!background} as="h3">
                {el.title}
              </Heading>
              <TextComponent>{el.description}</TextComponent>
            </Styled.GridElement>
          ))}
        </Styled.Grid>
      </Styled.Container>
    </SectionBackground>
  );
};

GridSection.propTypes = {
  background: P.bool,
  title: P.string.isRequired,
  description: P.string.isRequired,
  grid: P.arrayOf(
    P.shape({
      title: P.string.isRequired,
      description: P.string.isRequired,
    }),
  ).isRequired,
};
