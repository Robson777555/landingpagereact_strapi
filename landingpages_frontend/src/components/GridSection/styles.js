import styled, { css } from 'styled-components';
import { Title as HeadingContainer } from '../Heading/styles';
import { Container as TextComponent } from '../TextComponent/styles';
export const Container = styled.div`
  /* eslint-disable-next-line no-unused-vars */
  ${({ theme }) => css`
   font-family: ${theme.fonts.family.default};
   ${TextComponent} {
    margin-bottom: ${theme.spacings.xhuge};
   }
    
  `}
`;

export const Grid = styled.div`
  /* eslint-disable-next-line no-unused-vars */
  ${({ theme }) => css`
   font-family: ${theme.fonts.family.default};
   counter-reset: grid-counter; 
   display: grid;
   grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
   gap: ${theme.spacings.large};
  `}
`;

export const GridElement = styled.div`
  /* eslint-disable-next-line no-unused-vars */
  ${({ theme }) => css`
   font-family: ${theme.fonts.family.default};
   ${HeadingContainer} {
    position: relative;
    left: 5rem ;
   }

   ${HeadingContainer}::before {
    counter-increment: grid-counter;
    content: counter(grid-counter);
    position: absolute;
    font-size: 7rem;
    top: -3rem;
    left: -5rem;
    transform: rotate(5deg);
   }
    
  `}
`;
