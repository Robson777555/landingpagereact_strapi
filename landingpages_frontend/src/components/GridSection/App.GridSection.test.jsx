import { GridSection } from './GridSection';
import { renderTheme } from '../../styles/render-theme';
import { describe, it, expect } from '@jest/globals';
import mock from './mock';

describe('<GridSection />', () => {
  it('should render', () => {
    const { container } = renderTheme(<GridSection {...mock} />);
    expect(container).toMatchSnapshot();
  });
  it('should render grid section', () => {
    const { container } = renderTheme(
      <GridSection {...mock} background={undefined} />,
    );
    expect(container).toMatchSnapshot();
  });
});
