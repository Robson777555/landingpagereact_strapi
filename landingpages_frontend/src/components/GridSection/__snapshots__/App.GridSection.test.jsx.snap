// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridSection /> should render 1`] = `
.c4 {
  color: #0A1128;
  background: transparent;
  padding: 1rem;
  font-size: 6.4rem;
  text-transform: uppercase;
}

.c9 {
  color: #0A1128;
  background: transparent;
  padding: 1rem;
  font-size: 3.2rem;
  text-transform: none;
}

.c6 {
  font-size: 2.4rem;
}

.c2 {
  font-family: 'Open Sans',sans-serif;
}

.c2 .c5 {
  margin-bottom: 6.4rem;
}

.c7 {
  font-family: 'Open Sans',sans-serif;
  counter-reset: grid-counter;
  display: grid;
  grid-template-columns: repeat(auto-fill,minmax(280px,1fr));
  gap: 3.2rem;
}

.c8 {
  font-family: 'Open Sans',sans-serif;
}

.c8 .c3 {
  position: relative;
  left: 5rem;
}

.c8 .c3::before {
  counter-increment: grid-counter;
  content: counter(grid-counter);
  position: absolute;
  font-size: 7rem;
  top: -3rem;
  left: -5rem;
  transform: rotate(5deg);
}

.c0 {
  font-family: 'Open Sans',sans-serif;
  background: #FFFFFF;
  color: #0A1128;
  min-height: 100vh;
  display: flex;
  align-items: center;
}

.c1 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
}

@media (max-width:768px) {
  .c4 {
    font-size: 4.0rem;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <h2
          class="c3 c4"
        >
          My grid
        </h2>
        <div
          class="c5 c6"
        >
          Esse site é para estudos de react e técnicas de programação
        </div>
        <div
          class="c7"
        >
          <div
            class="c8"
          >
            <h3
              class="c3 c9"
            >
              Teste 1
            </h3>
            <div
              class="c5 c6"
            >
              Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Vestibulum tortor quam, feugiat vitae, ultricies eget, tempor sit amet, ante. Donec eu libero sit amet quam egestas semper. Aenean ultricies mi vitae est. Mauris placerat eleifend leo.
            </div>
          </div>
          <div
            class="c8"
          >
            <h3
              class="c3 c9"
            >
              Teste 2
            </h3>
            <div
              class="c5 c6"
            >
              Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Vestibulum tortor quam, feugiat vitae, ultricies eget, tempor sit amet, ante. Donec eu libero sit amet quam egestas semper. Aenean ultricies mi vitae est. Mauris placerat eleifend leo.
            </div>
          </div>
          <div
            class="c8"
          >
            <h3
              class="c3 c9"
            >
              Teste 3
            </h3>
            <div
              class="c5 c6"
            >
              Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Vestibulum tortor quam, feugiat vitae, ultricies eget, tempor sit amet, ante. Donec eu libero sit amet quam egestas semper. Aenean ultricies mi vitae est. Mauris placerat eleifend leo.
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<GridSection /> should render grid section 1`] = `
.c4 {
  color: #0A1128;
  background: transparent;
  padding: 1rem;
  font-size: 6.4rem;
  text-transform: uppercase;
}

.c9 {
  color: #0A1128;
  background: transparent;
  padding: 1rem;
  font-size: 3.2rem;
  text-transform: none;
}

.c6 {
  font-size: 2.4rem;
}

.c2 {
  font-family: 'Open Sans',sans-serif;
}

.c2 .c5 {
  margin-bottom: 6.4rem;
}

.c7 {
  font-family: 'Open Sans',sans-serif;
  counter-reset: grid-counter;
  display: grid;
  grid-template-columns: repeat(auto-fill,minmax(280px,1fr));
  gap: 3.2rem;
}

.c8 {
  font-family: 'Open Sans',sans-serif;
}

.c8 .c3 {
  position: relative;
  left: 5rem;
}

.c8 .c3::before {
  counter-increment: grid-counter;
  content: counter(grid-counter);
  position: absolute;
  font-size: 7rem;
  top: -3rem;
  left: -5rem;
  transform: rotate(5deg);
}

.c0 {
  font-family: 'Open Sans',sans-serif;
  background: #FFFFFF;
  color: #0A1128;
  min-height: 100vh;
  display: flex;
  align-items: center;
}

.c1 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
}

@media (max-width:768px) {
  .c4 {
    font-size: 4.0rem;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <h2
          class="c3 c4"
        >
          My grid
        </h2>
        <div
          class="c5 c6"
        >
          Esse site é para estudos de react e técnicas de programação
        </div>
        <div
          class="c7"
        >
          <div
            class="c8"
          >
            <h3
              class="c3 c9"
            >
              Teste 1
            </h3>
            <div
              class="c5 c6"
            >
              Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Vestibulum tortor quam, feugiat vitae, ultricies eget, tempor sit amet, ante. Donec eu libero sit amet quam egestas semper. Aenean ultricies mi vitae est. Mauris placerat eleifend leo.
            </div>
          </div>
          <div
            class="c8"
          >
            <h3
              class="c3 c9"
            >
              Teste 2
            </h3>
            <div
              class="c5 c6"
            >
              Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Vestibulum tortor quam, feugiat vitae, ultricies eget, tempor sit amet, ante. Donec eu libero sit amet quam egestas semper. Aenean ultricies mi vitae est. Mauris placerat eleifend leo.
            </div>
          </div>
          <div
            class="c8"
          >
            <h3
              class="c3 c9"
            >
              Teste 3
            </h3>
            <div
              class="c5 c6"
            >
              Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Vestibulum tortor quam, feugiat vitae, ultricies eget, tempor sit amet, ante. Donec eu libero sit amet quam egestas semper. Aenean ultricies mi vitae est. Mauris placerat eleifend leo.
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
